ui.frontend/src/main/webpack/site/js/components/searchInput.ts

```diff
@@ -39,6 +45,60 @@ const removeSuggestionList = (searchContainer: HTMLDivElement) => {
   }
 }
 
+const buildSuggestionElements = ({
+  results,
+  regexp,
+  query,
+  searchInput,
+  searchContainer,
+  suggestionDropdown,
+}: {
+  results: String[]
+  regexp: RegExp
+  query: string
+  searchInput: HTMLInputElement
+  searchContainer: HTMLDivElement
+  suggestionDropdown: Element
+}): void => {
+  const searchButtonElement = document.getElementsByClassName(
+    'saas-container_button',
+  )?.[0] as HTMLButtonElement
```

Comment:
shouldn't it be `as HTMLButtonElement | undefined` instead?

`searchButtonElement` could be undefined if `getElementsByClassName` doesn't return anything.